File: app/src/main/java/cdreyfus/xebia_henri_potier/basket/promotion/Minus.java
package cdreyfus.xebia_henri_potier.basket.promotion;

public class Minus extends CommercialOffer {
    private int value;

    public Minus(int value) {
        this.type = "minus";
        this.value = value;
    }

    public float applyMinus(float basketValue){
        return basketValue - value;
    }

    public int getValue() {
        return value;
    }


    @Override
    public float applyOffer(float basketValue) {
        return applyMinus(basketValue);
    }
}

File: app/src/main/java/cdreyfus/xebia_henri_potier/basket/promotion/Minus.kt
package cdreyfus.xebia_henri_potier.basket.promotion

class Minus(val value: Int) : CommercialOffer() {

    init {
        this.type = "minus"
    }

    private fun applyMinus(basketValue: Float): Float {
        return basketValue - value
    }


    override fun applyOffer(basketOffer: Float): Float {
        return applyMinus(basketOffer)
    }
}
